Lib/test/test_py_compile.py

```diff
@@ -98,6 +98,25 @@ def test_bad_coding(self):
         self.assertFalse(os.path.exists(
             importlib.util.cache_from_source(bad_coding)))
 
+    def test_source_date_epoch(self):
+        testtime = 123456789
+        orig_sde = os.getenv("SOURCE_DATE_EPOCH")
+        os.environ["SOURCE_DATE_EPOCH"] = str(testtime)
+        py_compile.compile(self.source_path, self.pyc_path)
+        if orig_sde:
+            os.environ["SOURCE_DATE_EPOCH"] = orig_sde
+        else:
+            del os.environ["SOURCE_DATE_EPOCH"]
+        self.assertTrue(os.path.exists(self.pyc_path))
+        self.assertFalse(os.path.exists(self.cache_path))
+        statinfo = os.stat(self.source_path)
+        self.assertEqual(statinfo.st_mtime, testtime)
+        f = open(self.pyc_path, "rb")
+        f.read(4)
+        timebytes = f.read(4) # read timestamp from pyc header
+        f.close()
+        self.assertEqual(timebytes, (testtime).to_bytes(4, 'little'))
```

Comment:
I would prefer to decode timebytes to retrieve the integer, instead of the opposite.